fms2_io/readme.md

```diff
@@ -0,0 +1,603 @@
+- [read_data](https://github.com/NOAA-GFDL/FMS/blob/main/fms2_io/include/domain_read.inc)
+  - The io_root pe reads the data and sends it to the other pes.
+
+#### 2. Unstructured Domain non-restart read/writes
+
+```F90
+use fms2_io_mod,        only: FmsNetcdfUnstructuredDomainFile_t, register_field, register_axis, unlimited
+use fms2_io_mod,        only: open_file, close_file, write_data
+use mpp_domains_mod,    only: domainug, center
+
+type(FmsNetcdfUnstructuredDomainFile_t) :: fileobj        !< Fms2_io domain decomposed fileobj
+real, dimension(:,:)        :: variable_data  !< Variable data in the unstructured domain
+type (domainug)             :: domain         !< 2d mpp domain
+character(len=8)            :: dim_names(3)   !< Array of dimension names
+
+dim_names(1) = "ucdim"
+dim_names(2) = "zaxis_1"
+dim_names(3) = "Time"
+
+!> Create domain !<
+!> Create an io_domain !<
+!> Create an unstructured domain !<
+
+if (open_file(fileobj, "filename", "overwrite", domain, is_restart=.true.)) then
+  call register_axis(fileobj, dim_names(1))
+  call register_axis(fileobj, dim_names(2), dimsize)
+  call register_axis(fileobj, dim_names(3), unlimited)
+
+  call register_field(fileobj, 'variable_name', 'variable_type', dim_names)
+  call write_data(fileobj, 'variable_name', variable_data)
+  call close_file(fileobj)
+endif
+```
+Difference from writing restarts:
+- `open_file` does not have is_restart=.true., so the `write_restart` functionality cannot be used'
+- [register_field](https://github.com/NOAA-GFDL/FMS/blob/b9fc6515c7e729909e59a0f9a1efc6eb1d3e44d1/fms2_io/fms_netcdf_unstructured_domain_io.F90#L178-L179) is basically a wrapper for `nf90_def_var` which just adds the variable metadata to the file.
+  - "variable_type" is a string which indicates the type you want the variable to be written as. The acceptable values are "int", "int64", "double", "float", and "char"
+- [write_data](https://github.com/NOAA-GFDL/FMS/blob/main/fms2_io/include/unstructured_domain_write.inc)
+  - The ranks send their data to the io_root pe. The io_root pe receives the data and writes it to the file.
+
+Similarly for reads:
+```F90
+if (open_file(fileobj, "filename", "read", domain, is_restart=.true.)) then
+  call register_axis(fileobj, dim_names(1))
+  call write_data(fileobj, 'variable_name', variable_data)
+  call close_file(fileobj)
+endif
+```
+- `register_axis` is required for the dimensions that are on the unstructured domain, so the code knows which dimensions and therefore variables are in the unstructured domain
+- `register_field` is not required because the code can get the dimensions information from the file.
+- [read_data](https://github.com/NOAA-GFDL/FMS/blob/main/fms2_io/include/unstructured_domain_read.inc)
+  - The io_root pe reads the data and sends it to the other pes.
+
+#### 3. Non-domain decomposed read/writes
+
+Writing non-domain decomposed files can be accomplished similarly to with domain decomposed files:
```

Comment:
Writing non-domain-decomposed files ~~can be accomplished similarly to with~~ is similar to writing domain-decomposed files: